sql/core/src/main/java/org/apache/spark/sql/sources/v2/writer/DataWriterFactory.java

```diff
@@ -48,6 +48,9 @@
    *                      same task id but different attempt number, which means there are multiple
    *                      tasks with the same task id running at the same time. Implementations can
    *                      use this attempt number to distinguish writers of different task attempts.
+   * @param epochId An opaque identifier for the current subset of data being operated on. Intended
```

Comment:
`A monotonically increasing id for streaming queries that are split in to discrete periods of execution.  For queries that execute as a single batch, this id will always be zero.`?

Reply:
changed, good idea